source/docs/software/kinematics-and-odometry/swerve-drive-kinematics.rst

```diff
@@ -303,14 +303,12 @@ By recording a set of swerve module states using :ref:`NetworkTables <docs/softw
 
         void Periodic() {
           // Periodically send a set of module states
-          swervePublisher.Set(
-            std::vector{
-              frontLeftState,
-              frontRightState,
-              backLeftState,
-              backRightState
-            }
-          );
+          std::vector states_vec = {m_frontLeft.GetState(), m_frontRight.GetState(),
+                            m_rearLeft.GetState(), m_rearRight.GetState()};
+         std::span<frc::SwerveModuleState, 4> states(states_vec.begin(),
+                                              states_vec.end());
+
+         publisher.Set(states);
```

Comment:
It shouldn't be necessary to explicitly create a std::span.  It should work to just pass states_vec directly to the publisher, as there's an implicit conversion from std::vector to std::span.

Reply:
It should also be possible to simply use double-braces here, although I've not tested it:
```c++
swervePublisher.Set(
            {{
              frontLeftState,
              frontRightState,
              backLeftState,
              backRightState
            }}
          );
```